pandas/io/common.py

```diff
@@ -198,6 +198,10 @@ def get_filepath_or_buffer(
     """
     filepath_or_buffer = stringify_path(filepath_or_buffer)
 
+    # use binary mode when converting path-like object to file-object
+    if isinstance(filepath_or_buffer, str) and mode is not None and "b" not in mode:
```

Comment:
tnx 👍 much appreciated 👍

Reply:
that didn't yet fix it, I forgot that I also need to return the updated mode. The next commit will hopefully fix that.

Reply:
It should be working now :) except the utf-16 issue and the `test_bad_stream_exception` test-case